server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1164,21 +1161,12 @@ public String innerGetRaw(final Settings settings) {
 
         @Override
         public Settings get(Settings settings) {
+            // TODO should we be checking for deprecations here?
             Settings byPrefix = settings.getByPrefix(getKey());
             validator.accept(byPrefix);
             return byPrefix;
         }
 
-        @Override
-        public boolean exists(Settings settings) {
```

Comment:
This override isn't necessary anymore with `exits` delegating to the key. Also, this check included secure setting keys, which is inconsistent with the default `exists`.